src/wavm/openmp.cpp

```diff
@@ -758,50 +817,9 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
 // REDUCTION
 // ---------------------------------------------------
 
-// Allowed reduce return values.
-// The single thread value is only used when there's one thread.
-// Multi threads signifies that multiple threads are operating
-// We don't currently support atomic reduce, but it signifies that the final
-// reduction must be performed atomically.
-enum ReduceReturnValue
-{
-    SINGLE_THREAD = 0,
-    MULTIPLE_THREADS = 1,
-    ATMOIC_REDUCE = 2,
-};
-
-/*
- * This method should return one of the reduce return values as detailed above.
- * AFAICT most of the arguments are not needed in our case.
- *
- * The reduction is performed by the master thread, so we must ensure that the
- * other threads have finished before this takes place.
- *
- * To interrogate where intermediate results are stored, we can look at the
- * reduce data and reduce func.
- *
- * The reduceFunc provides the reduction operation on two operands and returns
- * the result in its first argument. The function is of type:
- * void(*)(void *lhsData, void *rhsData)
- *
- * For more info, see OpenMP source at
- * https://github.com/llvm/llvm-project/blob/main/openmp/runtime/src/kmp_csupport.cpp
- */
 int reduceFinished()
 {
-    auto logger = faabric::util::getLogger();
-    auto ctx = threads::getOpenMPContext();
-
-    if (ctx.level->numThreads == 1) {
-        return SINGLE_THREAD;
-    }
-
-    // Notify the master thread that we've done our reduction
-    if (ctx.threadNumber != 0) {
-        ctx.level->masterWait(ctx.threadNumber);
-    }
-
-    return MULTIPLE_THREADS;
```

Comment:
This was all unnecessary and part of the bug causing different results between single and multi-threaded execution.